### ISSUE TITLE
Classification supervisée (basée sur les clusters)

### ISSUE THREAD

**Issue by SalimM21** [SalimM21/laboratoire-biom-dical-project#5]
- [x] Préparation des données pour la classification:
--> Définir la variable cible y à partir de la colonne Cluster.
--> Définir X à partir des variables sélectionnées.
--> Diviser les données en ensemble d’entraînement et de test à l’aide de train_test_split.
--> Gérer le déséquilibre des classes avec des techniques de sur-échantillonnage ou sous-échantillonnage (RandomOverSampler, UnderSampler via imblearn).

- [x] Entraînement de plusieurs modèles de classification
--> Tester les algorithmes suivants :
          1. Random Forest Support
          2. Vector Machine (SVM)
          3. Gradient Boosting Decision
          4. Tree Régression Logistique

- [x] Évaluation des modèles
--> Utiliser des métriques appropriées à la classification telles que :
          1. Matrice de confusion
          2. Précision
          3. Rappel
          4. F1-score

- [x] Validation croisée
--> Évaluer la robustesse des modèles sur différentes partitions du jeu de données en utilisant la validation croisée (cross-validation).

- [x] Optimisation des hyperparamètres
--> Utiliser des techniques comme GridSearchCV ou RandomizedSearchCV pour affiner les hyperparamètres des modèles et améliorer leurs performances.

- [x] Sélection et sauvegarde du meilleur modèle
          1. Comparer les performances des différents modèles.
          2. Choisir le modèle le plus performant.
          3. Sauvegarder le modèle final dans un fichier (ex. : model.pkl).
